README.md

```diff
@@ -4,6 +4,30 @@ One repo to finally have a clear, objective gRPC benchmark with code for everyon
 
 Contributions are most welcome!
 
+# Goal
+
+The goal of this benchmark is to compare the performance and resource usage of various gRPC libraries across different programming languages and technologies. To achieve that, a minimal protobuf contract is used to not pollute the results with other concepts (e.g. performances of hash maps) and to make the implementations simple.
+
+That being said, the service implementations should **NOT** take advantage of that and keep the code generic and maintainable. No inline assembly or other, language specific tricks / hacks. What does generic mean? One should be able to easily adapt the existing code to some fundamental use cases (e.g. having a thread-safe hash map on server side to provide values to client given some key).
+
+# You decide what is better
+
+Although in the end results are sorted according to the number of requests served, one should go beyond and look at the resource usage - perhaps one implementation is slightly better in terms of raw speed but uses three times more CPU to achieve that. Maybe it's better to take the first one if you're running on a Raspberry Pi and want to get the most of it. Maybe it's better to use the latter in a big server with 32 CPUs because it scales. It all depends on your use case. This benchmark is created to help people make an informed decision (and get ecstatic when their favourite technology seems really good, without doubts).
+
+# Metrics
+
+We try to provide some metrics to make this decision easier:
+
+* req/s - the number of requests the service was able to successfully serve
+* average latency, and 90/95/99 percentiles - time from sending a request to receiving the response
+* average CPU, memory - average resource usage during the benchmark, as reported by `docker stats`
+
+# What this benchmark does NOT take into account
+
+1. Completness of the gRPC library. We test only basic unary RPC at the moment. This is the most common service method which may be enough for some business use cases, but not for the others. When you're happy about the results of some technology, you should check out it's documentation (if it exists) and decide yourself is it production-ready.
```

Comment:
`Completeness`

Reply:
`if is it production-ready`.. I think
also we can leave out the documentation remark - some motivated people would just go and read the code anyway :)